src/server/rest/v1/service/SiteService.ts

```diff
@@ -392,6 +393,20 @@ export default class SiteService {
     site.name = filteredRequest.name;
     site.companyID = filteredRequest.companyID;
     if (Utils.objectHasProperty(filteredRequest, 'public')) {
+      if (!filteredRequest.public) {
+        const publicChargingStation = await ChargingStationStorage.getPublicChargingStationInSite(req.tenant, {
```

Comment:
No need to create a specific method in Storage for this you can call the method right away.